docs/adr/0011-operators.md

```diff
@@ -0,0 +1,153 @@
+
+## Decision Drivers
+
+- The definition of a `cluster-scope` application at [ADR 0010][] does not provide enough guidance for OLM.
+- Various OLM resources can be used in a way which effects the whole cluster or in a way that limits the effects to a single namespace.
+- Various OLM resources have permission restrictions or behave as a singletons within a namespace.
+- Some operators are not deployed via OLM, yet they behave as operators across all namespaces. Placement of those manifests is also disputed.
+- How should we handle meta-operators which operate across all namespaces and deploy other operators that operate across all namespaces?
+
+## Considered Options
+
+1. Deploy all operator-related resources within `cluster-scope` application.
+2. Require all namespaced resources to be deployed within individual applications, outside of `cluster-scope`.
+3. Find balance between 1. and 2. Deploy resources which have effect on a cluster level within `cluster-scope` app (also the namespace-scoped ones). The rest is deployed in individual applications.
+
+## Decision Outcome
+
+Chosen option: _"3. Find balance between 1. and 2."_, because it is desired to keep track of all resources which may effect other users and other namespaces i.e. operators provisioned to manage all cluster namespaces.
+
+Proposed decision defines strict rules for given resources. These rules are subject to future updates if needed, but the repositories and applications are required to consistently enforce them:
+
+### Operator Lifecycle Manager compatible operators
+
+[Operator Lifecycle Manager][] defines following resource kinds:
+
+- Cluster service version
+- Catalog source
+- Subscription
+- Install plan
+- Operator groups
+
+These are the set of guidelines we want to enforce on them:
+
+#### Cluster service version
+
+This resource is not meant to be deployed manually and is expected to be provided from a _Catalog source_. If there's really a need to deploy a custom _Cluster service version_, the resource has to live within the `cluster-scope` application.
+
+#### Catalog source
+
+It is uncommon for a _Catalog source_ to be declared manually. Operators are expected to be provisioned from the default catalogs and it is advised to rather publish the desired operator to a community catalog than deploying a custom _Catalog source_. If it is really inevitable to deploy an additional `CatalogSource` resource, this resource is required to be deployed to `openshift-marketplace` namespace and therefore it is required to declare this resource within the `cluster-scope` application.
```

Comment:
I'd like to suggest this rewrite:

> It is uncommon for a _Catalog source_ to be declared manually.  Operators are expected to be provisioned from the default catalogs and it is advised to publish the desired operator to a community catalog rather than deploying a custom _Catalog source_. If it is necessary to deploy an additional `CatalogSource` resource, this resource must be deployed to the `openshift-marketplace` namespace and therefore must be declared within the `cluster-scope` application.